dask_expr/reductions.py

```diff
@@ -151,26 +267,86 @@ def aggregate(cls, inputs, **kwargs):
         df = _concat(inputs)
         return func(df, **kwargs)
 
-    def __dask_postcompute__(self):
-        return toolz.first, ()
 
-    def _divisions(self):
-        return [None, None]
+##
+## Templates
+##
+
+
+class MapReduceTemplate:
```

Comment:
On first reading I don't understand this class.  It confuses me.  

Maybe that's alright, but it's a sign that others will likely be confused as well.

Reply:
This was an experiment that I consider a failure. I shared some thoughts here: https://github.com/mrocklin/dask-expr/pull/65#discussion_r1183989786